mindsdb/integrations/handlers/writer_handler/writer_handler.py

```diff
@@ -0,0 +1,110 @@
+from typing import Dict, Optional
+
+import pandas as pd
+
+from mindsdb.integrations.libs.base import BaseMLEngine
+from mindsdb.utilities.log import get_log
+
+from .ingest import Ingestor
+from .question_answer import QuestionAnswerer
+from .settings import (
+    DEFAULT_EMBEDDINGS_MODEL,
+    USER_DEFINED_MODEL_PARAMS,
+    ModelParameters,
+)
+
+# these require no additional arguments
+
+logger = get_log(logger_name=__name__)
+
+
+class WriterHandler(BaseMLEngine):
+    """
+    WriterHandler is a MindsDB integration with Writer API LLMs that allows users to run question answering
+    on their data by providing a question.
+
+    The User is able to provide data that provides context for the questions, see create() method for more details.
+
+    """
+
+    name = "writer"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    @staticmethod
+    def create_validation(target, args=None, **kwargs):
+        if "using" not in args:
+            raise Exception(
+                "Writer engine requires a USING clause! Refer to its documentation for more details."
+            )
+        else:
+            args = args["using"]
+
+        if "prompt_template" not in args:
+            raise Exception("Please provide a `prompt_template` for this engine.")
+
+    def create(
+        self,
+        target: str,
+        df: pd.DataFrame = pd.DataFrame(),
+        args: Optional[Dict] = None,
+    ):
+        """
+        Dispatch is running embeddings and storing in Chroma VectorDB, unless user already has embeddings persisted
+        see PERSIST_DIRECTORY in settings.py for default location of Chroma VectorDB indexes
```

Comment:
Probably best for a follow-up PR, but ideally users don't have to worry about this. Potential solution once the multiple-models-single-folder issue is fixed:

- Detect whether embeddings already exist for a given context
- If there aren't any, get and store them
- If there are, then user can ask to regenerate them with some other flag that would replace `get_embeddings`, e.g. `update_embeddings`

Reply:
yep, I had something along these lines planned but wanted to get something a bit quick and dirty out for the demo

Reply:
Awesome! And yes, 100% aligned re:demo and prioritization. I actually reviewed most of the PR afterwards because of that 😁